src/Orleans.Hosting.Kubernetes/KubernetesClusterAgent.cs

```diff
@@ -347,7 +347,7 @@ await foreach (var (eventType, pod) in pods.WatchAsync<V1PodList, V1Pod>(_shutdo
                         await Task.Delay(5000);
                     }
                 }
-                catch (Exception exception)
+                catch (Exception exception) when (!(_shutdownToken.IsCancellationRequested && (exception is TaskCanceledException || exception is OperationCanceledException)))
```

Comment:
```suggestion
                catch (Exception exception) when (!(_shutdownToken.IsCancellationRequested && exception is OperationCanceledException))
```
`TaskCanceledException` derives from `OperationCanceledException` https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcanceledexception?view=net-8.0